File: examples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/screens/DashboardKey.java
package com.zhuinden.navigationexamplefrag.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseFragment;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseKey;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class DashboardKey extends BaseKey {
    public static DashboardKey create() {
        return new AutoValue_DashboardKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new DashboardFragment();
    }
}

File: simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/screens/DashboardKey.kt
package com.zhuinden.simplestackexamplekotlinfragment.screens

import com.zhuinden.simplestackexamplekotlinfragment.core.navigation.BaseFragment
import com.zhuinden.simplestackexamplekotlinfragment.core.navigation.BaseKey
import kotlinx.android.parcel.Parcelize

/**
 * Created by Owner on 2017.11.13.
 */
@Parcelize
data class DashboardKey(private val placeholder: String = "") : BaseKey() { // generate reliable `toString()` for no-args data class
    override fun createFragment(): BaseFragment = DashboardFragment()
}
